### PR TITLE
Speech bubble refactor and Features

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Refactors speech bubbles. 

Basic idea inspired from TG, but their entire system doesn't work here.

What this PR does:

- Atoms that speak now will have a speech bubble
- Anything that speaks can have a defined speech bubble unique to it
- Adds in a lawyer's badge for IAA and Magistrate that allows for custom speech bubble
  - *OBJECTION!*
- Adds support for unique typing indicators as well

Code stuff
- Speech bubbles are no longer `spawn(0)`'d.

Generally allows for a more immersive customization that kinda sets the "tone" for how someone is speaking---also makes it easier to see when an atom is specifically talking as opposed to just making noise.

pics:
img
img
img

:cl: Fox McCloud
add: Robots, slimes, and other mobs now all have unique speech bubbles
add: Atoms that speak now have a speech bubble
add: Add's lawyer badges that allow IAA and Magistrate to---OBJECTION!
/:cl:

### CODE REVIEW

**Comment by TDSSS** [comment]
For a moment, I thought you were adding the speech floating over people's head thing.

But still, refactors good! 👍

**Comment by BakKeener** [comment]
I approve of the Lawyer Badge highly, too bad we hardly ever get trials.

File: code/modules/mob/typing_indicator.dm [hunk 1]
```diff
@@ -5,31 +5,32 @@ mob/var/typing
 mob/var/last_typed
 mob/var/last_typed_time
 
-GLOBAL_DATUM(typing_indicator, /image)
+GLOBAL_LIST_EMPTY(typing_indicator)
```
Comment (Fox-McCloud) [inline code/modules/mob/typing_indicator.dm:8]: This is now an associative list that acts as a cache.

**Comment by taukausanake** [comment]
We need speech bubbles with ears on them for when furries talk in their languages

**Comment by KasparoVy** [comment]
Any way to make it such that synthetic folks speak with the synth one-- like IPCs and such?

**Comment by Carthusia** [comment]
> For a moment, I thought you were adding the speech floating over people's head thing.
> 
> But still, refactors good! 👍

I hope we one day get the text over head thing as a toggleable option.

This is a cool little detail to add though

**Comment by Fox-McCloud** [comment]
>Any way to make it such that synthetic folks speak with the synth one-- like IPCs and such?

I could, I just wasn't sure what approach to take; should it be based on language? The mob itself? Making it language based would be kinda icky, to be honest (and not something I'm willing to do); making it based on the species is doable---I'm just not sure if that should be the case or not.

**Comment by ZomgPonies** [comment]
As long as slime people get the slime bubble. Could probably make a cool one for dionas too, but I dunno about vox or unathi or vulps.

**Comment by ZomgPonies** [comment]
At the very least, a cute speech bubble for diona nymphs!

**Comment by KasparoVy** [comment]
The approach Polaris took with the synth speech bubble is to check if the mob's visibly synthetic (exposed torso), and if they aren't, check if they have synthetic larynx, and failing both of those use regular speech bubble

Only normal & synth bubbles in that base tho!

**Comment by datlo** [comment]
Would be nice to get the synth bubble for greys with a translator if IPCs get it (they are meant to be talking in a text-to-speechy voice when they use one). I might do that myself in a future PR, though.

File: code/game/jobs/job/supervisor.dm [hunk 2]
```diff
@@ -262,9 +263,12 @@ GLOBAL_DATUM_INIT(captain_announcement, /datum/announcement/minor, new(do_newsca
 	glasses = /obj/item/clothing/glasses/hud/security/sunglasses/read_only
 	id = /obj/item/card/id/security
 	l_pocket = /obj/item/laser_pointer
-	r_pocket = /obj/item/flash
+	r_pocket = /obj/item/clothing/accessory/lawyers_badge
```
Comment (AffectedArc07) [inline code/game/jobs/job/supervisor.dm:266]: **OBJECTION**

Shouldnt this spawn on the lawyers jumpsuit itself?

**Comment by MRgaty** [comment]
> I could, I just wasn't sure what approach to take; should it be based on language? The mob itself? Making it language based would be kinda icky, to be honest (and not something I'm willing to do); making it based on the species is doable---I'm just not sure if that should be the case or not.

I mean a robot speaking chinese would still sound like a robot :p

File: code/game/jobs/job/supervisor.dm (hunk 2 above)
Comment (datlo) [reply code/game/jobs/job/supervisor.dm:266]: This'd be annoying if you start with a custom jumpsuit in loadout, imo its fine to be in inventory

**Comment by Spacemanspark** [comment]
Debatable, actually. It wouldn't be difficult to assume that IPCs/ cyborgs can speak normally.

**Comment by taukausanake** [comment]
I feel like it should be based on language to be honest. Granted you aren't going to have too many cases where a specific species will talk a language not native to them but I think I'd like to see that accounted for. If anything this is more fluff for characters so might as well fluff it out more

**Comment by Carthusia** [comment]
I mean, species kind of makes sense more than just language, no matter what language your speaking as a vox, your never going to get rid of that bird like accent

**Comment by taukausanake** [comment]
I think I remember if you got a slime limb though you could technically speak blubberish? Maybe that was a closed feature. And I feel like a Vox could overcome their accent if they really wanted to, it's just most people want to play them as literal bird brains

**Comment by Carthusia** [comment]
By accent, i dont mean the fact most vox have a distaste and disinterest in fully learning galactic common, i mean more their voice itself.

It's going to be REALLY hard to hide that kind of voice type.

Slimes sound goopy and sticky

Vox sound shriek and bird like

Unathi sound ssssnakelike

IPCs do/can sound robotic/normal

Diona sound creaky and or chirpy

Kidan sound clicky

Plasmamen sound gaspy

Drask sound oooooomy

and Humans sound like evolved monkeys who actually still think that digital watches are a pretty neat idea

**Pull request merged**